edx-solutions/xblock-drag-and-drop-v2
# Clean up studio edit template

**Description**:

This PR cleans up the Studio edit template and associated JS code to:

- Use `field.display_name` values for field labels instead of hardcoding/duplicating the names in the template.
- Directly display help text under associated input field instead of `title` and `<span class="sr">`.
- Replace usage of HTML `id` attributes with `class`-es. In cases where that is not possible, ensure that the `id` attributes are globally unique.
- Use `<button>` elements instead of `<a>` elements for buttons.
- Replace input placeholders with help texts visible to everyone.

**Screenshots**:

Before:

<img width="960" alt="screen shot 2016-08-04 at 17 53 48" src="https://cloud.githubusercontent.com/assets/32585/17397985/1c4c2a36-5a6d-11e6-9521-86a0551a1fc3.png">

After:

<img width="965" alt="screen shot 2016-08-04 at 17 52 52" src="https://cloud.githubusercontent.com/assets/32585/17397996/246f5558-5a6d-11e6-94c4-94e4c2bac7ca.png">

**JIRA ticket**: https://openedx.atlassian.net/browse/SOL-1990

**Dependencies**: None

**Discussions**: https://github.com/edx-solutions/xblock-drag-and-drop-v2/pull/77#discussion_r71034788

**Sandbox URLs** (installed version 9845083078d129009933cd8c8106b27c14e17242):

- LMS: http://dndv2-sandbox5.opencraft.hosting/
- Studio: http://studio-dndv2-sandbox5.opencraft.hosting/

You are probably mostly interested in the Studio edit functionality. Some DnDv2 blocks can be edited at: http://studio-dndv2-sandbox5.opencraft.hosting/container/block-v1:OpenCraftX+OC1+1+type@vertical+block@e3c62b19090948b8ad89d3386bff4435

**Testing Instructions**:

This PR does not implement any new functionality, so there is no specific feature to test.
The important thing is to ensure that the studio edit functionality continues to look and function correctly.

**Reviewers**:

- [x] OpenCraft: @bradenmacdonald
- [ ] ~~a11y: @edx/edx-accessibility~~ Scheduled combined review for all DnDv2.1 stories: [SOL-1978](https://openedx.atlassian.net/browse/SOL-1978)
